pkg/util/provider/machinecontroller/machine.go

```diff
@@ -190,6 +195,10 @@ func (c *controller) addNodeToMachine(obj interface{}) {
 		return
 	}
 
+	if _, annotationPresent := node.ObjectMeta.Annotations[machineutils.NotManagedByMCM]; annotationPresent {
```

Comment:
```suggestion
	 // If NotManagedByMCM annotation is present on node, don't process this node object
	if _, annotationPresent := node.ObjectMeta.Annotations[machineutils.NotManagedByMCM]; annotationPresent {
```
Just adding a comment for clarity for external contributors.